src/segmantic/i2i/translate.py

```diff
@@ -66,26 +62,132 @@ def load_cyclegan_generator(model_file_path: Path, gpu_ids: List[int] = []):
         use_dropout=False,
         init_type="normal",
         init_gain=0.02,
-        gpu_ids=gpu_ids,
+        gpu_ids=[device.index] if device.index else [],
     )
 
-    # get device name: CPU or GPU
-    device = (
-        torch.device("cuda:{}".format(gpu_ids[0])) if gpu_ids else torch.device("cpu")
-    )
-
-    # if you are using PyTorch newer than 0.4, you can remove str() on self.device
-    state_dict = torch.load(model_file_path, map_location=str(device))
+    state_dict = torch.load(model_file_path, map_location=device)
     if hasattr(state_dict, "_metadata"):
         del state_dict._metadata
     gen.load_state_dict(state_dict)
     return gen
 
 
-def translate(img: np.ndarray, model: torch.nn.Module, device: torch.device):
+def make_tiles(
+    size: Sequence[int],
+    tile_size: Tuple[int, int],
+    overlap: int = 0,
+    add_center_tile: bool = False,
+) -> List[Tuple[int, int]]:
+    """Break rectangular region into overlapping tiles of fixed size
+
+    Args:
+        size: shape of 2D image
+        tile_size: shape of 2D tiles
+        overlap: ensure overlap
+        add_center_tile: append centered tile
+
+    Returns:
+        List[Tuple[int, int]]: list of start indices of tiles
+    """
+    if size[0] < tile_size[0] and size[1] < tile_size[1]:
+        return [(0, 0)]
+
+    # ensure we don't go over end
+    fix_tile = lambda start: (
+        max(min(start[0], size[0] - tile_size[0]), 0),
+        max(min(start[1], size[1] - tile_size[1]), 0),
+    )
+
+    tile_indices = []
+    start = [0, 0]
+    while start[0] < size[0]:
+        start[1] = 0
+        while start[1] < size[1]:
+            tile_indices.append(fix_tile(start))
+            start[1] += tile_size[1]
+            if start[1] + 1 < size[1]:
+                start[1] -= overlap
+        start[0] += tile_size[0]
+        if start[0] + 1 < size[0]:
+            start[0] -= overlap
+
+    if add_center_tile:
+        delta = [int(max(s, t) - t) for s, t in zip(size, tile_size)]
```

Comment:
TIP: for ``delta`` if you use a generator instead of a list, you do not need to have the whole list in memory to evaluate ``any(.)``. 

generators is analogous in C++ to iterators. This is like passing a entire vector in memory vs passing begin() iterator


Just need to replace the ``[.]`` by ``(.)`` or even simply

```python
delta = ( int(max(s, t) - t) for s, t in zip(size, tile_size) )
if any(delta):
    ... 
    
# or even more one-liner
if any(int(max(s, t) - t) for s, t in zip(size, tile_size)):
   crop_low = ...
``` 

why do you cast to int? t and s are ints, right?

Reply:
casting is not necessary (probably added it to make mypy happy because the type hint was initially `tuple`)